File: 2348. Number of Zero-Filled Subarrays/Solution.py
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        res, n, prev = 0, len(nums), -1
        for i in range(n):
            if nums[i] != 0:
                res += (i - prev) * (i - prev - 1) // 2
                prev = i
        res += (n - prev) * (n - prev - 1) // 2
        return res

File: 2348. Number of Zero-Filled Subarrays/Solution.java
class Solution {
    public long zeroFilledSubarray(int[] nums) {
        long res = 0, n = nums.length, prev = -1;
        for (int i = 0; i < n; i++) {
            if (nums[i] != 0) {
                res += (i - prev) * (i - prev - 1) / 2;
                prev = i;
            }
        }
        res += (n - prev) * (n - prev - 1) / 2;
        return res;
    }
}